plenum/common/ledger_manager.py

```diff
@@ -86,6 +88,20 @@ def addLedger(self, iD: int, ledger: Ledger,
             verifier=MerkleVerifier(ledger.hasher)
         )
 
+    def request_ledger_status_if_needed(self, ledger_id):
+        ledgerInfo = self.getLedgerInfoByType(ledger_id)
+        nodes = [node for node in self.owner.nodeReg if node not in ledgerInfo.ledgerStatusOk]
+        self.owner.request_ledger_status_from_nodes(ledger_id, nodes)
+
+    def request_last_CP(self, ledgerId):
+        ledgerInfo = self.getLedgerInfoByType(ledgerId)
+        recvdConsProof = ledgerInfo.recvdConsistencyProofs
+        ledger_status = self.owner.build_ledger_status(ledgerId)
+        nodes = [frm for frm in self.owner.nodeReg if
+                 frm not in recvdConsProof and frm != self.owner.name]
```

Comment:
We should ask only the nodes that are definitely ahead of us and did not sent us ConsistencyProofs yet. (Now we also ask the nodes that have not even sent us LedgerStatuses.) To do this we should track the set of nodes which have already been asked for ConsistencyProofs and have not replied yet.

Reply:
Ignore this comment for this PR since we decided to re-ask messages only once and so we should re-ask even those nodes which even potentially may be ahead of us.